### PR TITLE
Add GUI test coverage for changes to Labels.show_selected_label

### LINKED ISSUE #4864: IndexError when changing selected label on labels layer
## 🐛 Bug

IndexError when changing selected label on labels layer.
This happens when changing very fast the label, having a layer containing 12 labels and selecting "show selected".

## To Reproduce

1. Have a labels layer (in my case 12 labels)
2. Select "Show selected"
3. Change the selected label by going from 1 upwards until it fails...

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
File /home/cnstt/PycharmProjects/napari-deepfinder/venv/lib/python3.9/site-packages/napari/_qt/layer_controls/qt_labels_controls.py:341, in QtLabelsControls.changeSelection(self=<napari._qt.layer_controls.qt_labels_controls.QtLabelsControls object>, value=19)
    333 def changeSelection(self, value):
    334     """Change currently selected label.
    335 
    336     Parameters
   (...)
    339         Index of label to select.
    340     """
--> 341     self.layer.selected_label = value
        value = 19
        self.layer = <Labels layer 'label_new5_binned' at 0x7f47458f1940>
        self = <napari._qt.layer_controls.qt_labels_controls.QtLabelsControls object at 0x7f46bbf4ec10>
    342     self.selectionSpinBox.clearFocus()
    343     self.setFocus()

File /home/cnstt/PycharmProjects/napari-deepfinder/venv/lib/python3.9/site-packages/napari/layers/labels/labels.py:623, in Labels.selected_label(self=<Labels layer 'label_new5_binned'>, selected_label=19)
    620     return
    622 self._selected_label = selected_label
--> 623 self._selected_color = self.get_color(selected_label)
        selected_label = 19
        self._selected_color = <class 'numpy.ndarray'> (4,) float64
        self = <Labels layer 'label_new5_binned' at 0x7f47458f1940>
    624 self.events.selected_label()
    626 # note: self.color_mode returns a string and this comparison fails,
    627 # so use self._color_mode

File /home/cnstt/PycharmProjects/napari-deepfinder/venv/lib/python3.9/site-packages/napari/layers/labels/labels.py:991, in Labels.get_color(self=<Labels layer 'label_new5_binned'>, label=19)
    989     col = self.colormap.map([0, 0, 0, 0])[0]
    990 else:
--> 991     val = self._raw_to_displayed(np.array([label]))
        label = 19
        self = <Labels layer 'label_new5_binned' at 0x7f47458f1940>
        [label] = [19]
        np = <module 'numpy' from '/home/cnstt/PycharmProjects/napari-deepfinder/venv/lib64/python3.9/site-packages/numpy/__init__.py'>
    992     col = self.colormap.map(val)[0]
    993 return col

File /home/cnstt/PycharmProjects/napari-deepfinder/venv/lib/python3.9/site-packages/napari/layers/labels/labels.py:959, in Labels._raw_to_displayed(self=<Labels layer 'label_new5_binned'>, raw=<class 'numpy.ndarray'> (1,) int64)
    954     image = self._color_lookup_func(raw_modified)
    955 elif (
    956     self.show_selected_label
    957     and self._color_mode == LabelColorMode.AUTO
    958 ):
--> 959     image = self._color_lookup_func(raw_modified, self._selected_label)
        raw_modified = <class 'numpy.ndarray'> (1,) int64
        self._color_lookup_func = <bound method Labels._lookup_with_index of <Labels layer 'label_new5_binned' at 0x7f47458f1940>>
        self = <Labels layer 'label_new5_binned' at 0x7f47458f1940>
        self._selected_label = 19
    960 elif (
    961     self.show_selected_label
    962     and self._color_mode == LabelColorMode.DIRECT
    963 ):
    964     selected = self._selected_label

File /home/cnstt/PycharmProjects/napari-deepfinder/venv/lib/python3.9/site-packages/napari/layers/labels/labels.py:831, in Labels._lookup_with_index(self=<Labels layer 'label_new5_binned'>, im=<class 'numpy.ndarray'> (1,) int64, selected_label=19)
    829     else:
    830         colors = np.zeros_like(self._all_vals)
--> 831         colors[selected_label] = low_discrepancy_image(
        selected_label = 19
        colors = <class 'numpy.ndarray'> (19,) float64
        self = <Labels layer 'label_new5_binned' at 0x7f47458f1940>
        self._seed = 0.5
    832             selected_label, self._seed
    833         )
    834         image = colors[im]
    835 else:

IndexError: index 19 is out of bounds for axis 0 with size 19

```

## Expected behavior

It should not fail.

## Environment

napari: 0.4.16
Platform: Linux-5.18.13-100.fc35.x86_64-x86_64-with-glibc2.34
System: Fedora Linux 35 (Workstation Edition)
Python: 3.9.13 (main, Jun 9 2022, 00:00:00) [GCC 11.3.1 20220421 (Red Hat 11.3.1-2)]
Qt: 5.15.2
PyQt5: 5.15.6
NumPy: 1.22.4
SciPy: 1.7.3
Dask: 2022.05.2
VisPy: 0.10.0

OpenGL:
- GL version: 4.6.0 NVIDIA 515.48.07
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- console: 0.0.4
- napari-deepfinder: 0.0.1
- napari-em-reader: 0.1.0
- napari-svg: 0.1.6
- scikit-image: 0.4.16

### LINKED ISSUE #5371: "show_selected_label" checkbox doesn't update when value is changed programmatically
## To Reproduce

1. Create a new Labels layer
2. Programmatically change `label_layer.show_selected_layer = True`
3. The UI doesn't update

## Expected behavior

Changes to the backend should update the UI

## Environment

napari main at 5fe0048cccc03db2b9afa2ecd30acdbeefb4bc40

## Additional context

Discovered while I was working on #4864. It wasn't possible to reproduce that particular issue programmatically without fixing this first.

### PR DESCRIPTION
**Pull request by kcpevey**
# References and relevant issues
Related to #4864 and #5371.

# Description
Adds test coverage for #5867 by checking that setting the layer property changes the checkbox state appropriately.

### CODE REVIEW

**Comment by kcpevey** [comment]
I'm not sure why this one test is failing? https://github.com/napari/napari/actions/runs/3550708891/jobs/5964336151

File: napari/_qt/layer_controls/_tests/test_qt_labels_layer.py [hunk 1]
```diff
@@ -22,6 +22,25 @@ def test_changing_layer_color_mode_updates_combo_box(qtbot):
...
+    and changing on the backend updates the UI"""
+    layer = Labels(_LABELS, color=_COLOR)
+    qtctrl = QtLabelsControls(layer)
+    qtbot.addWidget(qtctrl)
+
+    assert not qtctrl.selectedColorCheckbox.isChecked()
+    assert not layer.show_selected_label
+
+    # ensure setting UI checkbox changes backend
+    qtctrl.selectedColorCheckbox.setChecked(True)
```
Comment (Czaki) [inline napari/_qt/layer_controls/_tests/test_qt_labels_layer.py]:
```suggestion
    with qtbot.wait_signal(qtctrl.selectedColorCheckbox.stateChanged):
        qtctrl.selectedColorCheckbox.setChecked(True)
```
you need to ensure that the change state properly propagates.

Comment (kcpevey) [reply napari/_qt/layer_controls/_tests/test_qt_labels_layer.py]: ah. thank you!

**Comment by kcpevey** [comment]
Hmmm. @Czaki that didn't seem to fix it. Any other ideas?

**Comment by Czaki** [comment]
in code I see
```
 self.layer.show_selected_label = state == Qt.CheckState.Checked
```

So You may try to use [`check_params_cb`](https://pytest-qt.readthedocs.io/en/latest/signals.html#check-params-cb-parameter) to validate if the state is adequately set.

**Comment by kcpevey** [comment]
I'm still not sure why this is failing. I think perhaps there is some quirkiness with pyqt versioning? Not sure what else I can try?

**Comment by psobolewskiPhD** [comment]
Looks like it could be related to https://github.com/napari/napari/pull/5541
not sure why pyside6 passes though...

**Comment by andy-sweet** [comment]
Sorry for only responding now. I think the issues this fixes were also fixed by  #5541, #5867, and maybe #3308.

This PR does add test coverage and the test runs quickly, so I updated this PR to just add that. Also updated the PR description appropriately and made ready for review.

**Comment by andy-sweet** [comment]
@psobolewskiPhD : this is tiny now, so if you don't mind taking a look, it'd be much appreciated!

**Review by psobolewskiPhD — APPROVED** [review]
Makes sense to me!
Sorry this slipped @kcpevey !!
❤️

**Pull request merged**